src/main/java/com/google/sps/servlets/DestinationDataServlet.java

```diff
@@ -0,0 +1,147 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.Destination;
+import com.google.sps.data.LatLng;
+import com.google.sps.data.Riddle;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/* Returns a destination created from user submitted information */
+@WebServlet("/destination-data")
+public class DestinationDataServlet extends HttpServlet {
+
+  // Temporarily stores the destination created by the user
+  public List<Destination> destinations = new ArrayList<>();
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = request.getParameter("name");
+
+    LatLng location =
+        new LatLng.Builder()
+            .withLat(Double.parseDouble(request.getParameter("latitude")))
+            .withLng(Double.parseDouble(request.getParameter("longitude")))
+            .build();
+
+    String city = request.getParameter("city");
+    String description = request.getParameter("description");
+
+    Riddle riddle =
+        new Riddle.Builder()
+            .withPuzzle(request.getParameter("riddle"))
+            .withHint(request.getParameter("hint1"))
+            .withHint(request.getParameter("hint2"))
+            .withHint(request.getParameter("hint3"))
+            .build();
+
+    /* Retrieves the obscurity level chosen by the user as a List of Strings and converts the level to an Enum Obscurity value */
+    List<String> levels =
+        Arrays.stream(request.getParameterValues("obscurity"))
```

Comment:
this implies there can be more than 1 level of obsurity?

Reply:
I see the confusion. I will give it a more fitting name;

Reply:
I imagine you'd just want to use getParameterValue instead of getParameterValues, right?

Reply:
Yes actually that works better for this use case.

Reply:
there is no getParameterValues method for the HttpServletRequest type so I am only storing the first value in the array since there will only be one

Reply:
My mistake, the method is getParameter:
https://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#getParameter(java.lang.String)

Reply:
Ok I got that one to work.